### PR TITLE
[advanced editor] initialised advanced editor module

### PR DESCRIPTION
**Pull request by gastonche**
Initialised the advanced json editor module and did initial hooking up to the main module in order to start building the skeleton of the main module.

References #1.

### CODE REVIEW

File: src/js/index.js [hunk 1]
```diff
@@ -1,6 +1,16 @@
...
+	class netjsonEditor{
+		constructor({target}){
+			this.targetElement = $(target);
+			this.targetElement.hide();
+
+			this.advancedEditor = new advancedModule({target});
+		}
+	}
+	window.netjsonEditor = module.exports = netjsonEditor;
 })();
```
Comment (nemesisdesign) [inline src/js/index.js]: add a new line here, set your editor to automatically add new lines

Comment (gastonche) [inline src/js/index.js]: Okay. I will do that.

**Comment by gastonche** [comment]
Let me try to explain my approach. I have separated the advanced module and main module just like in my proposal. I am using ES6 classes  to build the modules and using browserify require to use one module in another. Using classes, to me, makes the code more modular, and gives me the possibility to easily separate concerns within the module, by using class methods, instead of functions that would affect the global scope. Also it is a lot easier to write cleaner and more understandable code with classes. I also believe the fact that classes have their  own scope, will avoid a lot of issues.

**Comment by nemesisdesign** [comment]
@gastonche yes sure I understood that, I see you are using some of the code of django-netjsonconfig, I wouldn't mind to start from scratch either. But up to you. We can clean up leater if needed.

**Comment by gastonche** [comment]
I already have the editor working so my work today is to clean and optimize the code. Then I will make the module more dynamic and customizable. This way I can now integrate it with the main module, hook the events correctly between the main module and the advanced editor module. From there I will be able to know what  next

**Comment by nemesisdesign** [comment]
@gastonche ok, proceed

**Comment by nemesisdesign** [comment]
@gastonche for the moment proceed as fast as possible. We'll start to clean up before the first evaluation.

**Pull request merged**